directory/glob.bzl

```diff
@@ -0,0 +1,86 @@
+# Copyright 2024 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Rules to filter files from a directory."""
+
+load(":providers.bzl", "DirectoryInfo")
+load(":utils.bzl", _directory_glob = "directory_glob")
+
+def _remove_exclude(files, exclude):
+    if not exclude:
+        return files
+    files = {k: None for k in files.to_list()}
+    for value in exclude:
+        files.pop(value, None)
+    return depset(files.keys())
```

Comment:
Might it be clearer to implement this as `return [file for file in files.to_list() where file not in set(exclude)]`? (I'm guessing you probably started with this and it didn't work for some reason, so I'm mainly curious what the blocker was.)